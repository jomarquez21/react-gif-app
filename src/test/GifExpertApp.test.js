import React from 'react';
import {shallow} from 'enzyme';
import {GifExpertApp} from './../GifExpertApp';
import '@testing-library/jest-dom';

describe('Test functional on `GifExpertApp`', () => {
  test('Deberia renderizar correctamente el component', () => {
    const component = shallow(<GifExpertApp defaultCategories={[]} />);

    expect(component).toMatchSnapshot();
  });

  test('Deberia renderizar una lista de categorias', () => {
    const categories = ['Naruto', 'Samurai x'];
    const component = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(component.find('GifGrid').length).toBe(categories.length);
  });
});
