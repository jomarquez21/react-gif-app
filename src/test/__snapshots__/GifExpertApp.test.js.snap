// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test functional on \`GifExpertApp\` Deberia renderizar correctamente el component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
